plone/app/linkintegrity/parser.py

```diff
@@ -30,17 +30,34 @@ def handle_starttag(self, tag, attrs):
             self.links.extend(search_attr('href', attrs))
         if tag == 'img':
             self.links.extend(search_attr('src', attrs))
+            self.links.extend(search_attr('srcset', attrs))
         if tag == 'source':
+            # Used within img/picture/audio/video tags
+            # to embed various sources of media.
             self.links.extend(search_attr('src', attrs))
             self.links.extend(search_attr('srcset', attrs))
+        if tag == 'audio':
+            # Embeds audio recordings.
+            self.links.extend(search_attr('src', attrs))
+        if tag == 'video':
+            # Embeds videos.
+            self.links.extend(search_attr('src', attrs))
+        if tag == 'iframe':
+            # Used to embed PDFs.
+            self.links.extend(search_attr('src', attrs))
```

Comment:
this - and above - is a bit lot of the same now.
What about defining two module globals with sets like `TAGS_WITH_SRC = {"img", "source", "audio", ...}` and one with `TAGS_WITH_SRCSET = {...}.and then have two ifs like `if tag in TAGS_WITH_SRC: ....} and same for the other set?

Reply:
You're right and it's done.